trustfall_core/src/frontend/mod.rs

```diff
@@ -1104,6 +1103,30 @@ where
         {
             // Processing a property.
 
+            // @fold is not allowed on a property
+            if connection.fold.is_some() {
```

Comment:
There is some code duplication around here, but at least the signatures I thought about would keep things highly coupled with the context of this method. So I kept the code duplication because I thought it was the cleaner version of the code. Let me know if you have ideas for making this better.

Reply:
This is totally fine. Not all duplication is bad, and in this case the duplicated code is fairly short and very clear. Anything less duplicated would be way more convoluted and less readable.